### DISCUSSION TITLE
Assign target to follow in "Framed" mode

### DISCUSSION THREAD

**Discussion by woyosensei** [ramokz/phantom-camera#223] [Question / Help Wanted] [answered]
Hello, everyone!
I apologise for such a basic question but how can I assign a target to follow in a Framed mode using code?
I have something I call "mission template" which is set of basic nodes I can use for any mission I am creating. In that template I have PhantomCamera3D, Camera3D and PhantomCameraHost as its child. The PhantomCamera3D is set to "Framed" mode but no target just yet. I have a method in the scirpt for that template called "playerSpawn" where I want to assign the "Follow Target" from PhantomCamera3D to a specific node in the player's scene once the player is spawned (i.e. after dying or when everything on the scene is loaded, like during ProtonScatter init which sometimes can take a while to load). So my question is:
how can I assign a "Follow Target" using code?

If you need more explanation please let me know.
Thank you very much in advance for any help.

**Comment by ramokz** [comment, marked as answer]
Not a basic question at all!

Normally, you can assign a follow target using the setter method: `pcam.set_follow_target_node(your_follow_target)`

However, due to bugs with this particular `follow mode`, it will not work particularly well when done so via code. Partly because the distance offset is only being calculated in the editor and the reframing bugs out unless the target is within the dead when you assign the follow target. You can get around it by manually offsetting the `PCam3D`'s position and rotating it towards where the follow target will be right before you assign the follow target.

It's one of the things planned to be properly addressed in #161

**Reply by woyosensei** [reply]
Thank you for your response.
I found a solution. I have a Marker3D node in that template scene which serves as a spawner point (more or less, its purpose is irrelevant for this case) and I assign a PhantomCamera3D `follow_target` to this Marker3D. When the scene is loaded then I swap `follow_target` to the new created player scene via code using the setter method you've mentioned above. This way it works exactly the way I want, at least for now :) Thank you very much for pointing me into the right direction.
Best Regards.
